scripts/sanity_chk/sanitylib.py

```diff
@@ -902,16 +905,18 @@ def handle(self):
                 else:
                     proc.terminate()
                     proc.kill()
-                    self.returncode = proc.returncode
-            else:
-                self.returncode = proc.returncode
 
+            self.returncode = proc.returncode
```

Comment:
if put `self.returncode = proc.returncode` here, the returncode in case
```
                if os.path.exists(self.pid_fn):
                    qemu_pid = int(open(self.pid_fn).read())
                    try:
                        os.kill(qemu_pid, signal.SIGKILL)
                    except ProcessLookupError:
                        pass
                    proc.wait()
                    self.returncode = 0
```
will be overwrite.

Reply:
isnt this the point? Why are you passing 0 return code in case of a timeout?

Reply:
there is a comment about this case, it intents to handle QEMU doesn't handle SIGTERM signal timely and correctly in sanitycheck environment(for example sample/Philosopher case), send SIGKILL signal to QEMU process and let sanitycheck judge the result by console output. 
```
#sometimes QEMU can't handle SIGTERM signal correctly
#in that case kill -9 QEMU process directly and leave
#sanitycheck judge testing result by console output
```

Reply:
can you suggest a fix then, since you were the one who introduced this bug?

Reply:
will do.

Reply:
the fix will be like below:
```
diff --git a/scripts/sanity_chk/sanitylib.py b/scripts/sanity_chk/sanitylib.py
index a083c34b64..b2fa4d11f2 100644
--- a/scripts/sanity_chk/sanitylib.py
+++ b/scripts/sanity_chk/sanitylib.py
@@ -898,7 +898,10 @@ class QEMUHandler(Handler):
                     except ProcessLookupError:
                         pass
                     proc.wait()
-                    self.returncode = 0
+                    if harness.state == "passed":
+                        self.returncode = 0
+                    else:
+                        self.returncode = proc.returncode
                 else:
                     proc.terminate()
                     proc.kill()
```